sc-locked-asset-factory/src/lib.rs

```diff
@@ -106,10 +116,13 @@ pub trait LockedAssetFactory:
         let locked_token_id = self.locked_asset_token_id().get();
         require!(token_id == locked_token_id, "Bad payment token");
 
-        let attributes = self.get_attributes(&token_id, token_nonce)?;
+        let unlock_schedule = self.get_unlock_schedule_for_sft_nonce(token_nonce).unwrap();
         let current_block_epoch = self.blockchain().get_block_epoch();
-        let unlock_amount =
-            self.get_unlock_amount(&amount, current_block_epoch, &attributes.unlock_milestones);
+        let unlock_amount = self.get_unlock_amount(
```

Comment:
I think sft.start_epoch should be sent for get_unlock_amount

Reply:
Fixed

Reply:
why? need to calculate the unlock with regard to the current epoch (or at least current month epoch)

Reply:
yeah. not more needed.